File: apps/api/src/lib/ids.ts
import { randomBytes } from 'node:crypto';

/** Prefixed, sortable-enough ids: out-m4k1x9ab. Matches the frontend reducer's newId(). */
export function newId(prefix: string): string {
  return `${prefix}-${Date.now().toString(36)}${randomBytes(3).toString('hex')}`;
}

File: apps/api/internal/httpx/ids.go
package httpx

import (
	"crypto/rand"
	"encoding/hex"
	"strconv"
	"time"
)

// NewID mirrors the frontend reducer's newId(): a readable prefix plus enough entropy.
func NewID(prefix string) string {
	buf := make([]byte, 3)
	_, _ = rand.Read(buf)
	return prefix + "-" + strconv.FormatInt(time.Now().UnixMilli(), 36) + hex.EncodeToString(buf)
}
